integrated_channels/moodle/client.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""
+Client for connecting to Moodle.
+"""
+
+import json
+from urllib.parse import urlencode
+
+import requests
+
+from django.apps import apps
+
+from integrated_channels.integrated_channel.client import IntegratedChannelApiClient
+
+
+class MoodleAPIClient(IntegratedChannelApiClient):
+    """
+    Client for connecting to Moodle.
+    Transmits learner and course metadata.
+
+    Required configuration to access Moodle:
+    - wsusername and wspassword:
+        - Web service user and password created in Moodle. Used to generate api tokens.
+    - Moodle base url.
+        - Customer's Moodle instance url.
+        - For local development just `http://localhost` (unless you needed a different port)
+    """
+
+    def __init__(self, enterprise_configuration):
+        """
+        Instantiate a new client.
+
+        Args:
+            enterprise_configuration (MoodleEnterpriseCustomerConfiguration): An enterprise customers's
+            configuration model for connecting with Moodle
+        """
+        super(MoodleAPIClient, self).__init__(enterprise_configuration)
+        self.config = apps.get_app_config('moodle')
+
+    def create_course_completion(self, user_id, payload):  # pylint: disable=unused-argument
+        pass
+
+    def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-argument
+        pass
+
+    def get_course_id(self, key):
+        """
+        Gets the Moodle course id (because we cannot update/delete without it).
+        """
+        params = {
+            'wstoken': self.enterprise_configuration.api_token,
+            'wsfunction': 'core_course_get_courses_by_field',
+            'field': 'shortname',
+            'value': key,
+            'moodlewsrestformat': 'json'
+        }
+        response = requests.get(
+            self.enterprise_configuration.moodle_base_url,
+            params=params
+        )
+
+        return json.loads(response.text)['courses'][0]['id']
+
+    def create_content_metadata(self, serialized_data):
+        """
+        The below assumes the data is dict/object.
+        Format should look like:
+        {
+          courses[0][shortname]: 'value',
+          courses[0][fullname]: 'value',
+          [...]
+          courses[1][shortname]: 'value',
+          courses[1][fullname]: 'value',
+          [...]
+        }
+        """
+
+        serialized_data['wstoken'] = self.enterprise_configuration.api_token
```

Comment:
can this method also reuse the _post internal method?

Reply:
I wouldn't bother reviewing client code at this stage (kind of surprised you're reviewing this now suddenly) as Jeff branched off of this PR for all the client code.

Reply:
Your comment did prompt me to go back and try to understand why this was not using the post method, and I don't really remember why. I think I just missed it honestly (I was back and forth initially on using a unified post method). But Jeff's PR does use the `_post` method.